core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -445,8 +445,9 @@ class ProducerStateManager(val topicPartition: TopicPartition,
             lastMapOffset = lastSnapOffset
             return
           } catch {
-            case e: CorruptSnapshotException =>
-              error(s"Snapshot file at ${file.getPath} is corrupt: ${e.getMessage}")
+            case e: Exception =>
```

Comment:
Wouldn't it be better to catch all exceptions here, as the patch does right now? If we can't read the file for any reason, best to delete it and either fallback on the next snaphsot or rebuild the producer state from the log. It seems like the safest approach here.